sonic_installer/main.py

```diff
@@ -334,6 +337,32 @@ def install(url, force):
         # copy directories and preserve original file structure, attributes and associated metadata
         run_command("cp -ar /etc/sonic /host/old_config")
 
+    # Check Latest BIOS availability
+    try:
+        import sonic_platform.chassis
+        component_name = "BIOS"
+        chassis = sonic_platform.chassis.Chassis()
+
+        next_image = get_next_image()
+        squashfs_path = HOST_PATH + '/' + \
+                next_image.replace(IMAGE_PREFIX, IMAGE_DIR_PREFIX) + \
+                '/' + 'fs.squashfs'
+        if squashfs_path:
+            tmpdir = tempfile.mkdtemp()
+            run_command(
+                'mount ' + squashfs_path + ' ' + tmpdir + \
+                ' -t squashfs -o loop')
+            status = chassis.install_component_firmware(component_name, tmpdir)
+
+            if not status:
+                click.echo("BIOS upgrade failed!")
```

Comment:
if bios upgrade failed, we should umount the squashfs as well.

Reply:
- We check status variable here only to indicate BIOS upgrade failure.
- unmounting squashfs is performed irrespective of success/failure. You can see the implementation below:
- https://github.com/Azure/sonic-utilities/blob/d708fc5117a2d1c80e246d90ea09ff047ecb7d50/sonic_installer/main.py#L360

Reply:
- We check status variable here only to indicate BIOS upgrade failure.
- unmounting squashfs is performed irrespective of success/failure. You can see the implementation below:
- https://github.com/Azure/sonic-utilities/blob/d708fc5117a2d1c80e246d90ea09ff047ecb7d50/sonic_installer/main.py#L360